servicetalk-transport-api/src/main/java/io/servicetalk/transport/api/ConnectionInfo.java

```diff
@@ -25,6 +25,33 @@
  * Provides information about a connection.
  */
 public interface ConnectionInfo {
+    /**
+     * String representation of the current connection information.
+     *
+     * @return String representation of the current connection information.
+     */
+    @Override
+    String toString();
+
+    /**
+     * String representation of an identifier for this connection (can be globally non-unique).
+     * <p>
+     * Note: this identifier is a string representation of an ID assigned by underlying implementation of the
+     * connection. Whether it's globally unique or not depends on that implementation. It's not recommended to use this
+     * identifier as a map key for storing connection related data. It can be used for logging purposes to correlate
+     * events happening on this connection with other logs or events related to the same instance. If necessary,
+     * uniqueness can be ensured by using a combination of the current identifier with {@link #localAddress()} and
+     * {@link #remoteAddress()}.
+     *
+     * @return String representation of an identifier for this connection (can be globally non-unique).
+     */
+    default String connectionId() {
+        final long hashCode = System.identityHashCode(this);
+        final int high = (int) (hashCode >>> 32);
+        final int low = (int) (hashCode & 0xFFFFFFFFL);
+        return String.format("0x%08x", high ^ low);
```

Comment:
I'm not sure what's going on here. I believe `System.identityHashCode(..)` returns an `int`. That means the `high` part will always be 0 and you could just do `String.format("0x%08x", System.identityHashCode(this))`;